lib/iris/aux_factory.py

```diff
@@ -784,20 +758,51 @@ def dependencies(self):
                     depth_c=self.depth_c, nsigma=self.nsigma, zlev=self.zlev)
 
     def _derive(self, sigma, eta, depth, depth_c,
-                zlev, shape, nsigma_slice):
+                zlev, nsigma, coord_dims_func):
+        # Calculate the index of the 'z' dimension in the inputs.
+        # Get the cube dimension...
+        i_levels_cubedim, = coord_dims_func(self.dependencies['zlev'])
+        # ...subtract any previous dimensions not used in the calculation.
+        derived_cubedims = self.derived_dims(coord_dims_func)
+        i_levels_dim = i_levels_cubedim - sum(
+            i_dim not in derived_cubedims
+            for i_dim in range(i_levels_cubedim))
```

Comment:
@pp-mo Would you buy into renaming `i_levels_cubedim` to `zlev_dim`, and `i_levels_dim` to `zlev_index` or `zlev_offset`.

There's a lot going on here, and using `i_levels_...` feels (to me) like its one level if indirection that could be avoided ...